lib/misc-types.d.ts

```diff
@@ -0,0 +1,48 @@
+import { Transaction } from './transaction';
```

Comment:
I don't like adding a declaration file that doesn't exist in the source library.

Reply:
Do you have a better suggestion? These types are used across multiple files, copy/pasting them into each file would make this a PITA to maintain.

Reply:
Well in the source code the reason why these options are all possible on e.g. the associations is that under the hood they will all call into `Model` methods, so that's where I would put it.